### ISSUE TITLE
Add configurable learning-rate scheduling with cosine annealing and linear warmup

### ISSUE THREAD

**Issue by abhaygoudannavar** [mllam/neural-lam#282]
ARModel.configure_optimizers() currently returns a bare torch.optim.AdamW with a fixed learning rate for the entire training run:

``` python
# neural_lam/models/ar_model.py  L201-205
def configure_optimizers(self):
    opt = torch.optim.AdamW(
        self.parameters(), lr=self.args.lr, betas=(0.9, 0.95)
    )
    return opt
```
This means neural-lam trains with a constant lr=1e-3 for all 200 epochs, regardless of model size or dataset. There is no option to configure any learning rate schedule — no warmup, no decay, nothing.

PyTorch Lightning natively supports returning {"optimizer": ..., "lr_scheduler": ...} from 
configure_optimizers, but this capability is currently unused.

## Why it matters
- Training stability: Graph neural networks with multi-step autoregressive unrolling (where gradients flow through ar_steps forward passes) are highly sensitive to learning rate. A warmup phase prevents early gradient explosions before the model's residual connections stabilize.
- Convergence quality: Cosine annealing has been shown to yield better minima than constant LR in virtually all modern deep learning training. For weather prediction, this directly translates to lower forecast RMSE at longer lead times.
- Scientific reproducibility: Published NWP-ML papers specify exact LR schedules. Without scheduler support, neural-lam users cannot reproduce published results and must resort to manual hacks:
1.GraphCast (Lam et al., 2023): 1000-step linear warmup → cosine decay to 0
2.Pangu-Weather (Bi et al., 2023): cosine annealing with warmup
3.GenCast (Price et al., 2024): linear warmup → cosine decay
4.FourCastNet (Pathak et al., 2022): cosine annealing with warmup
- Zero performance cost: This is a pure configuration change — no additional compute overhead.

**Comment by leifdenby** [comment]
You are absolutely right that this is something we need, this is however being discussed in https://github.com/mllam/neural-lam/issues/97 and associated PR :)
